dwave/system/temperatures.py

```diff
@@ -0,0 +1,596 @@
+    problems exploiting embeddings, or with many coupled variables, may freeze out at 
+    different values of s or piecemeal). Large problems may have slow dynamics at small 
+    values of s, so :math:`A(s)` cannot be ignored as a contributing factor to the distribution.
+
+    Note that for QPU solvers this temperature estimate applies to problems
+    submitted with no additional scaling factors (sampling with ``auto_scale = False``). 
+    If ``auto_scale=True`` (default) additional scaling factors must be accounted for. 
+
+    Args:
+        freezeout_B (float):
+             :math:`B(s^*)`, the problem Hamiltonian energy scale at freeze-out.
+
+        temperature (float):
+            :math:`T`, the physical temperature of the quantum computer.
+        
+        units_B (string, optional, 'GHz'):
+            Units in which ``freezeout_B`` is specified. Allowed values:
+            'GHz' (Giga-Hertz) and 'J' (Joules).
+
+        units_T (string, optional, 'mK'):
+            Units in which the ``temperature`` is specified. Allowed values:
+            'mK' (milli-Kelvin) and 'K' (Kelvin).
+
+    
+    Returns:
+        float : The effective (unitless) temperature. 
+    
+    Examples:
+
+       This example uses the 
+       published parameters <https://docs.dwavesys.com/docs/latest/doc_physical_properties.html>
+       for the Advantage_system4.1 QPU solver as of November 22nd 2021: 
+       :math:`B(s=0.612) = 3.91` GHz , :math:`T = 15.4`mK.
+       
+       >>> from dwave.system.temperatures import freezeout_effective_temperature
+       >>> T = freezeout_effective_temperature(freezeout_B = 3.91, temperature = 15.4)
+       >>> print('Effective temperature at single qubit freeze-out is', T)  # doctest: +ELLIPSIS
+       Effective temperature at single qubit freeze-out is 0.164...
+    
+    See also:
+
+        The function :class:`~dwave.system.temperatures.fast_effective_temperature` 
+        estimates the temperature for single-qubit Hamiltonians, in approximate
+        agreement with estimates by this function at reported single-qubit 
+        freeze-out values :math:`s^*` and device physical parameters.
+    '''
+    
+    #Convert units_B to Joules
+    if units_B == 'GHz':
+        h = 6.626068e-34 #J/Hz
+        freezeout_B = freezeout_B *h
+        freezeout_B *= 1e9
+    elif units_B == 'J':
+        pass
+    else:
+        raise ValueException("Units must be 'J' (Joules) "
+                             "or 'mK' (milli-Kelvin)")
+    
+    if units_T == 'mK':
+        temperature = temperature * 1e-3
+    elif units_T == 'K':
+        pass
+    else:
+        raise ValueException("Units must be 'K' (Kelvin) "
+                             "or 'mK' (milli-Kelvin)")
+    kB = 1.3806503e-23 # J/K
+    
+    return 2*temperature*kB/freezeout_B
+
+def fast_effective_temperature(sampler=None, num_reads=None, seed=None,
+                               h_range=[-1/6.1,1/6.1], sampler_params=None,
```

Comment:
Default argument value [shouldn't be mutable](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).